src/XR/features/WebXRLightEstimation.ts

```diff
@@ -0,0 +1,366 @@
+            internalTexture.format = Constants.TEXTUREFORMAT_RGBA;
+            internalTexture.generateMipMaps = false;
+            internalTexture.type = this.options.reflectionFormat !== 'srgba8' ? Constants.TEXTURETYPE_HALF_FLOAT : Constants.TEXTURETYPE_UNSIGNED_BYTE;
+            internalTexture.samplingMode = Constants.TEXTURE_NEAREST_SAMPLINGMODE;
+            internalTexture.width = this.ReflectionCubeMapTextureSize;
+            internalTexture.height = this.ReflectionCubeMapTextureSize;
+            internalTexture._cachedWrapU = Constants.TEXTURE_WRAP_ADDRESSMODE;
+            internalTexture._cachedWrapV = Constants.TEXTURE_WRAP_ADDRESSMODE;
+            internalTexture._hardwareTexture = new WebGLHardwareTexture(lp, this._getCanvasContext() as WebGLRenderingContext);
+            this._reflectionCubeMap._texture = internalTexture;
+            this._reflectionCubeMap._texture.isReady = true;
+            this._xrSessionManager.scene.markAllMaterialsAsDirty(Constants.MATERIAL_TextureDirtyFlag);
+            this._reflectionCubeMap.hasAlpha = true;
+
+            this.onReflectionCubeMapUpdatedObservable.notifyObservers(this._reflectionCubeMap);
+        }
+    }
+
+    /**
+     * attach this feature
+     * Will usually be called by the features manager
+     *
+     * @returns true if successful.
+     */
+    public attach(): boolean {
+        if (!super.attach()) {
+            return false;
+        }
+
+        const reflectionFormat = this.options.reflectionFormat ?? (this._xrSessionManager.session.preferredReflectionFormat || "srgba8");
+        this.options.reflectionFormat = reflectionFormat;
+        this._xrSessionManager.session.requestLightProbe({
+            reflectionFormat
+        }).then((xrLightProbe: XRLightProbe) => {
+            this._xrLightProbe = xrLightProbe;
+            if (!this.options.disableCubeMapReflection) {
+                if (!this._reflectionCubeMap) {
+                    this._reflectionCubeMap = new BaseTexture(this._xrSessionManager.scene);
+                    this._reflectionCubeMap.isCube = true;
+                    this._reflectionCubeMap.coordinatesMode = Constants.TEXTURE_CUBIC_MODE;
+                    if (this.options.setSceneEnvironmentTexture) {
+                        this._xrSessionManager.scene.environmentTexture = this._reflectionCubeMap;
+                    }
+                }
+                this._xrLightProbe.addEventListener('reflectionchange', this._updateReflectionCubeMap);
+            }
+        });
+
+        return true;
+    }
+
+    /**
+     * detach this feature.
+     * Will usually be called by the features manager
+     *
+     * @returns true if successful.
+     */
+    public detach(): boolean {
+        const detached = super.detach();
+
+        if (this._xrLightProbe !== null) {
+            this._xrLightProbe.removeEventListener('reflectionchange', this._updateReflectionCubeMap);

```

Comment:
Shouldn't you test also for `!this.options.disableCubeMapReflection` before removing the event listener, as the attach code is testing for this condition before adding the event listener?

Reply:
Sure! removeEventListener is a no-op if it doesn't exist, but there is really no need to do that here, you are totally right.